proxy.js

```diff
@@ -14,7 +14,7 @@ exports.callEndpoint = function(appId, appSecret, endpoint, query, token, callba
   };
 
   request(options, function (err, response, body) {
-    if (err) {
+    if (err || response.statusCode != 200) {
```

Comment:
Separate error and response, since if status code is not 200, err is null, thus the error message we are going to get will not be informative.

After the error check add a different if which checks the status code and calls the callback with its own error object, which also contains the status code in the error message